src/WinRT.Runtime/Marshalers.cs

```diff
@@ -1374,16 +1374,33 @@ struct MarshalInterface<T>
     {
 #if NET
         [DynamicallyAccessedMembers(
-            DynamicallyAccessedMemberTypes.PublicFields | 
-            DynamicallyAccessedMemberTypes.NonPublicFields | 
+            DynamicallyAccessedMemberTypes.PublicFields |
             DynamicallyAccessedMemberTypes.PublicNestedTypes |
             DynamicallyAccessedMemberTypes.PublicMethods)]
 #endif
-        private static readonly Type HelperType = typeof(T).GetHelperType();
-        private static FieldInfo _ObjField;
+        private static Type _HelperType;
 
-        // Either a Func<T, IObjectReference> (JIT) or a boxed Guid (NativeAOT)
-        private static object _CreateMarshalerOrIid;
+#if NET
+        [DynamicallyAccessedMembers(
+            DynamicallyAccessedMemberTypes.PublicFields |
+            DynamicallyAccessedMemberTypes.PublicNestedTypes |
+            DynamicallyAccessedMemberTypes.PublicMethods)]
+#endif
+        private static Type HelperType => _HelperType ??= typeof(T).GetHelperType();
+
+        private static Func<T, IObjectReference> _CreateMarshaler;

```

Comment:
Why is this an explicit field? That was done on purpose (reusing the same field, I mean) to avoid one delegate generic instantiation on NAOT. Is this done now because we'll need that `IID` field on both paths?

Reply:
right, IID is used as part of `BindCreateMarshaler` which is what was also stored here.  Do we want to instead make this `object` to avoid the generic instantiation in the delegate?  We can also in theory use the helper type to get the IID in that code path as it is not a NAOT code path.

Reply:
Yeah we could just make that an `object` to be fully pay for play here 🤔